specification/search/data-plane/Azure.Search/stable/2024-07-01/searchservice.json

```diff
@@ -0,0 +1,11548 @@
+          }
+        ]
+      },
+      "description": "The compression method used for indexing and querying."
+    },
+    "VectorSearchVectorizer": {
+      "type": "object",
+      "discriminator": "kind",
+      "properties": {
+        "name": {
+          "externalDocs": {
+            "url": "https://docs.microsoft.com/rest/api/searchservice/Naming-rules"
+          },
+          "type": "string",
+          "description": "The name to associate with this particular vectorization method.",
+          "x-nullable": false
+        },
+        "kind": {
+          "$ref": "#/definitions/VectorSearchVectorizerKind",
+          "description": "The name of the kind of vectorization method being configured for use with vector search.",
+          "x-nullable": false
+        }
+      },
+      "required": [
+        "name",
+        "kind"
+      ],
+      "description": "Specifies the vectorization method to be used during query time."
+    },
+    "AzureOpenAIVectorizer": {
+      "type": "object",
+      "x-ms-discriminator-value": "azureOpenAI",
+      "allOf": [
+        {
+          "$ref": "#/definitions/VectorSearchVectorizer"
+        }
+      ],
+      "properties": {
+        "azureOpenAIParameters": {
+          "x-ms-client-name": "AzureOpenAIParameters",
+          "$ref": "#/definitions/AzureOpenAIParameters",
+          "description": "Contains the parameters specific to Azure OpenAI embedding vectorization."
+        }
+      },
+      "description": "Specifies the Azure OpenAI resource used to vectorize a query string."
+    },
+    "AzureOpenAIParameters": {
+      "type": "object",
+      "properties": {
+        "resourceUri": {
+          "type": "string",
+          "format": "uri",
+          "description": "The resource URI of the Azure OpenAI resource."
+        },
+        "deploymentId": {
+          "type": "string",
+          "description": "ID of the Azure OpenAI model deployment on the designated resource."
+        },
+        "apiKey": {
+          "type": "string",
+          "description": "API key of the designated Azure OpenAI resource."
+        },
+        "authIdentity": {
+          "$ref": "#/definitions/SearchIndexerDataIdentity",
+          "x-nullable": true,
+          "description": "The user-assigned managed identity used for outbound connections."
+        },
+        "modelName": {
+          "$ref": "#/definitions/AzureOpenAIModelName",
+          "description": "The name of the embedding model that is deployed at the provided deploymentId path."
+        }
+      },
+      "description": "Specifies the parameters for connecting to the Azure OpenAI resource."
+    },
+    "AzureOpenAIModelName": {
+      "type": "string",
+      "enum": [
+        "text-embedding-ada-002",
+        "text-embedding-3-large",
+        "text-embedding-3-small",
+        "experimental"
```

Comment:
From Review: Instead of "experimental", since this is a string enum consider accepting null and then not performing any additional validation